hub/core/chunk_engine/write.py

```diff
@@ -187,9 +204,10 @@ def write_bytes(
     for chunk in chunk_generator:
         if extend_last_chunk:
             chunk_name = last_chunk_name
-            last_chunk_bytearray = bytearray(last_chunk)
-            last_chunk_bytearray.extend(chunk)
-            chunk = bytes(last_chunk_bytearray)
+
+            last_chunk += chunk  # type: ignore
+            chunk = memoryview(last_chunk)
```

Comment:
Are we writing memoryviews into the cache, or does it eventually get unpacked back out into the bytes? 

We still have this conversion to bytearray each time, right? It feels like we should just by using bytearrays everywhere. Why would we care about the immutability that `bytes` provides?

Reply:
true, maybe i can just create 1 bytearray for the chunk and pre-allocate it for a full chunk? then use a memoryview on it

Reply:
I'll approve for now, but make sure this gets addressed if it's still a bottleneck